src/OpenLoco/Graphics/Gfx.cpp

```diff
@@ -971,17 +1005,34 @@ namespace OpenLoco::Gfx
         string_id stringId,
         const void* args)
     {
-        registers regs;
-        regs.al = colour.u8();
-        regs.bx = stringId;
-        regs.bp = width;
-        regs.cx = x;
-        regs.dx = y;
-        regs.esi = X86Pointer(args);
-        regs.edi = X86Pointer(&context);
-        call(0x00495224, regs);
+        char buffer[256];
+        StringManager::formatString(buffer, std::size(buffer), stringId, args);
+
+        _currentFontSpriteBase = Font::medium_bold;
+        auto wrapResult = wrapString(buffer, width);
+        auto breakCount = wrapResult.second;
 
-        return regs.dx;
+        // wrapString might change the font due to formatting codes
+        uint16_t lineHeight = 0; // _112D404
+        if (_currentFontSpriteBase <= Font::medium_bold)
+            lineHeight = 10;
+        else if (_currentFontSpriteBase == Font::small)
+            lineHeight = 6;
+        else if (_currentFontSpriteBase == Font::large)
+            lineHeight = 18;
+
+        _currentFontFlags = 0;
+        Ui::Point point = { x, y };
+        const char* ptr = buffer;
+
+        for (auto i = 0; ptr != nullptr && i <= breakCount; i++)
+        {
+            Gfx::drawString(context, point.x, point.y, colour, const_cast<char*>(ptr));
+            ptr = advanceToNextLine(ptr);
+            point.y += lineHeight;
```

Comment:
This is losing the colour used to paint. For example, the first line might switch to painting using a window colour, which the successive lines then do not use.

Reply:
It is losing colour as the original used to run the drawString function once to set the global colours then for the looping it passed in 0xFE for colour which is a special code to tell it not to set the global colours at the start.

Reply:
Can be fixed by adding the following on L1012:
```
Gfx::drawString(context, context.x, context.y, colour, "");
```
and modifying L1030 to be:
```
Gfx::drawString(context, point.x, point.y, AdvancedColour::FE(), const_cast<char*>(ptr));
```